### ISSUE TITLE
Promoted CLI Missing From API Reference Section

### ISSUE THREAD

**Issue by speediedan** [Lightning-AI/lightning#14071]
Thanks for your immensely valuable contributions to the ML research community!

I've noticed that since the [recent promotion of the cli](https://github.com/Lightning-AI/lightning/pull/13767) out of utilities, the new location is missing its reference in the API Reference:

```bash
python -msphinx.ext.intersphinx https://pytorch-lightning.readthedocs.io/en/latest/objects.inv | grep "pytorch_lightning.*.cli.LightningCLI "

pytorch_lightning.utilities.cli.LightningCLI api/pytorch_lightning.utilities.cli.html#pytorch_lightning.utilities.cli.LightningCLI
```
I'm submitting a PR shortly to fix this issue by simply adding the new api location (in addition to the deprecated existing one) so that [intersphinx references](https://finetuning-scheduler.readthedocs.io/en/stable/advanced/lr_scheduler_reinitialization.html#:~:text=dictionary%20via%20the-,LightningCLI,-%2C%20one%20could%20use) to the CLI can be updated to the new location and that the new location is properly reflected in the API Reference section:
```bash
python -msphinx.ext.intersphinx ../build/html/objects.inv | grep 'pytorch_lightning.*.cli.LightningCLI '

pytorch_lightning.cli.LightningCLI       api/pytorch_lightning.cli.LightningCLI.html#pytorch_lightning.cli.LightningCLI
pytorch_lightning.utilities.cli.LightningCLI api/pytorch_lightning.utilities.cli.html#pytorch_lightning.utilities.cli.LightningCLI
...
```
Thanks again!

cc @borda @rohitgr7 @carmocca @mauvilsa
